lib/execution_segment_test.go

```diff
@@ -379,4 +423,343 @@ func TestExecutionSegmentStringSequences(t *testing.T) {
+			require.NoError(t, err)
+			assert.Equal(t, ess.String(), ess2.String())
+		})
+	}
+}
+
+func TestSequenceLCD(t *testing.T) {
+	testCases := []struct {
+		seq string
+		lcd int64
+	}{
+		{seq: "0,0.3,0.5,0.6,0.7,0.8,0.9,1", lcd: 10},
+		{seq: "0,0.1,0.5,0.6,0.7,0.8,0.9,1", lcd: 10},
+		{seq: "0,0.2,0.5,0.6,0.7,0.8,0.9,1", lcd: 10},
+		{seq: "0,1/3,5/6", lcd: 6},
+		{seq: "0,1/3,4/7", lcd: 21},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(fmt.Sprintf("seq:%s", tc.seq), func(t *testing.T) {
+			ess, err := NewExecutionSegmentSequenceFromString(tc.seq)
+			require.NoError(t, err)
+			require.Equal(t, tc.lcd, ess.lcd())
+		})
+	}
+}
+
+func BenchmarkGetStripedOffsets(b *testing.B) {
+	var lengths = [...]int64{10, 100}
+	generateRandomSequence := func(n int64, r *rand.Rand) ExecutionSegmentSequence {
+		// try to randomly generate an executionsegmentsequence
+		var err error
+		var ess = ExecutionSegmentSequence(make([]*ExecutionSegment, n))
+		var numerators = make([]int64, n)
+		var denominator int64
+		for i := int64(0); i < n; i++ {
+			for numerators[i] == 0 {
+				numerators[i] = r.Int63n(n)
+				denominator += numerators[i]
+			}
+		}
+		ess[0], err = NewExecutionSegment(big.NewRat(0, 1), big.NewRat(numerators[0], denominator))
+		require.NoError(b, err)
+		for i := int64(1); i < n; i++ {
+			ess[i], err = NewExecutionSegment(ess[i-1].to, new(big.Rat).Add(big.NewRat(numerators[i], denominator), ess[i-1].to))
+			require.NoError(b, err, "%d", i)
+		}
+
+		return ess
+	}
+
+	const seed = 777
+	r := rand.New(rand.NewSource(seed))
+
+	for _, length := range lengths {
+		length := length
+		b.Run(fmt.Sprintf("length%d,seed%d", length, seed), func(b *testing.B) {
+			sequence := generateRandomSequence(length, r)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				segment := sequence[int(r.Int63())%len(sequence)]
+				et := NewExecutionTuple(segment, &sequence)
+				_, _, _ = et.GetStripedOffsets(segment)
+			}
+		})
+	}
+}
+
+func BenchmarkGetStripedOffsetsEven(b *testing.B) {
+	var lengths = [...]int64{10, 100, 1000}
+	generateSequence := func(n int64) ExecutionSegmentSequence {
+		var err error
+		var ess = ExecutionSegmentSequence(make([]*ExecutionSegment, n))
+		var numerators = make([]int64, n)
+		var denominator int64
+		for i := int64(0); i < n; i++ {
+			numerators[i] = 1 // nice and simple :)
+			denominator += numerators[i]
+		}
+		ess[0], err = NewExecutionSegment(big.NewRat(0, 1), big.NewRat(numerators[0], denominator))
+		require.NoError(b, err)
+		for i := int64(1); i < n; i++ {
+			ess[i], err = NewExecutionSegment(ess[i-1].to, new(big.Rat).Add(big.NewRat(numerators[i], denominator), ess[i-1].to))
+			require.NoError(b, err, "%d", i)
+		}
+
+		return ess
+	}
+
+	for _, length := range lengths {
+		length := length
+		b.Run(fmt.Sprintf("length%d", length), func(b *testing.B) {
+			sequence := generateSequence(length)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				segment := sequence[111233%len(sequence)]
```

Comment:
I wanted to not be the first or the last segment ... but maybe I should've gone for 555555 in order to get a more in the middle one ... 

Not really any good reason - I needed something to be able to benchmark the changes between versions, now that it will compute it for all segments either way I think it matters even less